File: Sesion 7/Marco teorico/11herencia_virtual_protected/11.py
class Figura:
    def dibujar(self):
        print("Dibujando Figura")

class Triangulo(Figura):
    def dibujar(self):
        print("Dibujando Triangulo")

class Rectangulo(Figura):
    def dibujar(self):
        print("Dibujando Rectangulo")

t = Triangulo()
r = Rectangulo()

t.dibujar()  
r.dibujar()  

File: Sesion 7/Marco teorico/11herencia_virtual_protected/11.java
import java.util.*;

class Figura {
    public void dibujar() {
        System.out.println("Dibujando Figura");
    }
}

class Triangulo extends Figura {
    @Override
    public void dibujar() {
        System.out.println("Dibujando Triangulo");
    }
}

class Rectangulo extends Figura {
    @Override
    public void dibujar() {
        System.out.println("Dibujando Rectangulo");
    }
}

public class Main {
    public static void main(String[] args) {
        Triangulo t = new Triangulo();
        Rectangulo r = new Rectangulo();

        t.dibujar(); 
        r.dibujar(); 
    }
}
